src/Process.php

```diff
@@ -110,13 +110,24 @@ public function start(LoopInterface $loop, $interval = 0.1)
             stream_set_blocking($pipe, 0);
         }
 
-        $loop->addPeriodicTimer($interval, function (Timer $timer) {
+        $periodicMonitor = function (Timer $timer) {
```

Comment:
How about moving the above anonymous function to a (public) `checkRunning()` (better names welcome) instead?

This way, the user would have more control over *when* to check the process status. After all, the consumer of this API will probably have a better understanding on *when* to consider a process dead.